x-pack/plugins/alerting/server/application/maintenance_window/methods/find/find_maintenance_windows.ts

```diff
@@ -9,17 +9,32 @@ import Boom from '@hapi/boom';
 import { MaintenanceWindowClientContext } from '../../../../../common';
 import { transformMaintenanceWindowAttributesToMaintenanceWindow } from '../../transforms';
 import { findMaintenanceWindowSo } from '../../../../data/maintenance_window';
-import type { FindMaintenanceWindowsResult } from './types';
+import type { FindMaintenanceWindowsResult, FindMaintenanceWindowsParams } from './types';
+import { findMaintenanceWindowsParamsSchema } from './schemas';
 
 export async function findMaintenanceWindows(
-  context: MaintenanceWindowClientContext
+  context: MaintenanceWindowClientContext,
+  params?: FindMaintenanceWindowsParams
 ): Promise<FindMaintenanceWindowsResult> {
   const { savedObjectsClient, logger } = context;
 
   try {
-    const result = await findMaintenanceWindowSo({ savedObjectsClient });
+    if (params) {
+      findMaintenanceWindowsParamsSchema.validate(params);
+    }
+  } catch (error) {
+    throw Boom.badRequest(`Error validating find maintenance windows data - ${error.message}`);
+  }
+
+  try {
+    const result = await findMaintenanceWindowSo({
+      savedObjectsClient,
+      ...(params ? { savedObjectsFindOptions: params } : {}),
```

Comment:
nit: Let's map explicitly the params to the `savedObjectsFindOptions`. Like

```
savedObjectsFindOptions: { page: params.page, perPage: params.perPage }
```

Reply:
I did like this, because  params can be true, but equal {}. `params.page` and `params.perPage` will be undefined as well. in that case you suggested it will be `savedObjectsFindOptions: { page: undefined, perPage: undefined }`, not savedObjectsFindOptions: { }
Fine by me, if you think it's better. Works fine either way.

Reply:
I think is nice to map because in the future we can accidentally slip options we do not like. By mapping, we are sure what we pass is what we want.